### PR TITLE
Change some mercurial/ hg.python.org references.

### PR DESCRIPTION
**Pull request by orsenthil**

### CODE REVIEW

**Comment by vstinner** [comment]
You may also update:

```
Lib/idlelib/help.html:93:<p><strong>Source code:</strong> <a class="reference external" href="https://hg.python.org/cpython/file/3.5/Lib/idlelib/">Lib/idlelib/</a></p>
Tools/README:46:(*) A generic benchmark suite is maintained separately at http://hg.python.org/benchmarks/
Tools/importbench/README:6:hg.python.org/benchmarks.
```

For benchmarks, please point to the *new* project: https://github.com/python/performance

By the way, you might use reST syntax for links to enhance formatting.

File: Doc/faq/general.rst [hunk 1]
```diff
@@ -222,7 +222,7 @@ releases are announced on the comp.lang.python and comp.lang.python.announce
 newsgroups and on the Python home page at https://www.python.org/; an RSS feed of
 news is available.
 
-You can also access the development version of Python through Mercurial.  See
+You can also access the development version of Python through Git.  See
 https://docs.python.org/devguide/faq.html for details.
```
Comment (Mariatta) [inline Doc/faq/general.rst]: Devguide URL should be updated to http://cpython-devguide.readthedocs.io.
Also, the FAQ has been removed from the Dev Guide, so the section you want is perhaps http://cpython-devguide.readthedocs.io/communication.html

Comment (zware) [reply Doc/faq/general.rst]: `https://docs.python.org/devguide` is still correct, but the lack of faq does need to be fixed.  Perhaps just remove the `/faq.html` part.

Comment (methane) [reply Doc/faq/general.rst]: I think one correct URL is better.
At least, https://docs.python.org/devguide doesn't switched to github version yet.

Comment (Mariatta) [reply Doc/faq/general.rst]: Update:
https://docs.python.org/devguide has been updated to github.

At the moment /faq.html is still alive, but will be removed soon (see https://github.com/python/devguide/issues/114)

Let's remove the reference to faq.html here :)

**Comment by orsenthil** [comment]
Thanks for the review comments, @haypo , @Mariatta . I've addressed them
- I saw only .rst page and restructured, the link
- We have an opportunity in another PR to .rst the other README's that we maintain the library directory. I'll let this PR just a changing references of hg.python.org to the github repo.

Please review again, if good to go, I am willing to try our merge process with merging this to master and cherrying pick to other active branches. 

Thanks!

**Review by berkerpeksag — APPROVED** [review]
Looks good to me. Note that you probably should use `needs backport to X.Y` labels instead of `cherry-pick for X.Y`.

**Comment by orsenthil** [comment]
Thanks @berkerpeksag  for your review.

**Comment by methane** [comment]
ref:
* #183 for 2.7
* #184 for 3.5
* #185 for 3.6

**Pull request merged**